cloud/deploys/bin/azure/bastion.sh

```diff
@@ -0,0 +1,72 @@
+#! /usr/bin/env bash
+
+#######################################
+# Get the ip for the bastion VM 
+# Arguments:
+#   1: the resource group name
+#######################################
+function bastion::get_vm_ip() {
+  echo $(az network public-ip show \
+    -g "${1}" \
+    -n "${1}-ip" \
+    --query "ipAddress" | tr -d '"')
+}
+
+#######################################
+# Get the postgres host for the psql server
+# Arguments:
+#   1: the resource group name
+#   2: the name of the database
+#######################################
+function bastion::get_postgres_host() {
+  echo $(az postgres server show \
+    -g "${1}" \
+    -n "${2}" \
+    --query "fullyQualifiedDomainName" | tr -d '"')
+}
+
+#######################################
+# Remove the ssh key we created after you connect 
+# Arguments:
+#   1: the key name to remove
+#######################################
+function bastion::remove_ssh_key() {
+  echo "rm {1}; rm ${1}.pub"
+}
+
+#######################################
+# Get the ssh command to connect to the bastion
+# Arguments:
+#   1: the ip of the vm you will connect to
+#   2: the key name to use to connect to
+#   3: command to run after ssh'ing
+#######################################
+function bastion::get_ssh_command() {
+  echo "ssh -i ${2} adminuser@${1} '${3}'"
+}
+
+#######################################
+# Update the bastion to allow ssh given a public key
+# Arguments:
+#   1: the resource group name
+#   2: the key name to use to connect to
+#######################################
+function bastion::update_bastion_ssh_keys() {
+  echo "az vm user update \
+    -u adminuser \
+    -g "${1}" \
+    -n "${1}-bstn-vm" \
+    --ssh-key-value \"$(< ${2}.pub)\""
+}
```

Comment:
question (blocking): in a previous iteration there were separate scripts for printing the commands one could run and actually running the commands, which is why I suggested echoing strings like this and using `eval`. If we don't need the former though I recommend not using `echo` / `eval` since it makes the code harder to read and reason about.

Is it still the intention to provide the commands in both ways?

Reply:
yeah i changed them so that just the get ssh command and the get postgres command are print/eval and the other ones are just executed!

Reply:
if i don't do the ssh commands in the echo eval it doesn't forward the std in/std out (which I think means something about the command is a lil off)

Reply:
okay altered this a decent amount, resolving as i think it should be fixed